### PR TITLE
Example/blog previous next navigation

### PR DESCRIPTION
**Pull request by spences10**
## Description
Add previous/next example made from the basic example: https://github.com/nozzle/react-static/pull/662#issuecomment-404926631

## Changes/Tasks

- [x] Changed code

## Motivation and Context

Example of how to use previous next in a blog

## Types of changes

- [ ] Refactoring/add tests (refactoring or adding test which isn't a fix or add a feature)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [ ] My changes have tests around them

### CODE REVIEW

**Comment by spences10** [comment]
How do I get to the beta branch?

There isn't a `beta` branch to select and I can't `npm i react-static@v6.0.0-beta.13`

**Comment by tannerlinsley** [comment]
Master branch

**Comment by spences10** [comment]
I updated from the master branch before submitting the PR 🙃

**Comment by spences10** [comment]
Will update on resolution of #684

**Comment by spences10** [comment]
I can't see what's different @tannerlinsley 

I have created a basic example with `npx react-static@next create` and selected the basic template, there's nothing I can see that's changed, the version is still @5.9.7 in the created example.

Would it be simpler if I wait for v6 to be released?

**Comment by danilobuerger** [comment]
@spences10 As you can see by going to https://www.npmjs.com/package/react-static and then clicking on the "Versions" tab, the Tag `6.0.0-beta.15` is marked as `next`. Maybe the update didn't work.

**Comment by spences10** [comment]
@danilobuerger rather than point out that I've done something wrong could you not offer anything helpful?

image

Like what the differences are? How to identify them, or something other than "you've done it wrong try again!"

**Comment by tannerlinsley** [comment]
😭 Looks like I totally dropped the ball on this one. Since then, v6 has been put in it's own branch, so if you choose to reopen this PR (I hope you do), you can point at that branch.

I will leave some notes in the files for what has changed.

File: examples/previous-next-navigation/static.config.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  getSiteData: () => ({
+    title: 'React Static',
+  }),
+  getRoutes: async () => {
+    const { data: posts } = await axios.get('https://jsonplaceholder.typicode.com/posts')
+    return [
+      {
+        path: '/',
+        component: 'src/containers/Home',
+      },
```
Comment (tannerlinsley) [inline examples/previous-next-navigation/static.config.js]: If the `Home.js` component is moved to `pages/index.js`, then this route definition isn't needed anymore 😄.

File: examples/previous-next-navigation/static.config.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      {
+        path: '/about',
+        component: 'src/containers/About',
+      },
```
Comment (tannerlinsley) [inline examples/previous-next-navigation/static.config.js]: Same here. The About.js component could be moved to `pages/about.js` and the route will be inferred.

File: examples/previous-next-navigation/static.config.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      {
+        path: '/blog',
+        component: 'src/containers/Blog',
+        getData: () => ({
+          posts,
+        }),
```
Comment (tannerlinsley) [inline examples/previous-next-navigation/static.config.js]: Similarly, you could move `Blog.js` to `pages/blog.js` and then remove the `component` line. You would however need to keep the `path` and `getData`, to properly tie the data to that route.

File: examples/previous-next-navigation/static.config.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+              path: `/blog/post/${post.id + 1}`
+            }, 
+            post
+          })
+        })) 
+      },
+      { 
+        is404: true,
+        component: 'src/containers/404'
+      }
```
Comment (tannerlinsley) [inline examples/previous-next-navigation/static.config.js]: `404.js` could be moved to `pages/404.js` and this entire route definition could be removed. Yay!

**Comment by tannerlinsley** [comment]
Looks like the branch has been deleted, so reopening will not work until the branch is restored. Or you can just open a new PR and refer to this one for notes?

**Comment by spences10** [comment]
> Since then, v6 has been put in it's own branch, so if you choose to reopen this PR (I hope you do), you can point at that branch.

@tannerlinsley there is no v6 branch, what should I submit a PR to?

image

**Comment by tannerlinsley** [comment]
The branch is located here: https://github.com/nozzle/react-static/tree/v6. I'm not sure why it's not showing up in your list there. 🤔

**Comment by spences10** [comment]
>The branch is located here: https://github.com/nozzle/react-static/tree/v6. I'm not sure why it's not showing up in your list there. 🤔

Because I was looking at my fork! 🙃 
image

Oh boy!

File: examples/previous-next-navigation/static.config.js (hunk 1 above)
Comment (spences10) [reply examples/previous-next-navigation/static.config.js]: Done

File: examples/previous-next-navigation/static.config.js (hunk 4 above)
Comment (spences10) [reply examples/previous-next-navigation/static.config.js]: Done

**Comment by spences10** [comment]
Resubmitted with https://github.com/nozzle/react-static/pull/798 @tannerlinsley 

Thanks

**Pull request closed without merging**